integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JMSInboundMessagePollingTestCase.java

```diff
@@ -111,6 +133,45 @@ public void destroy() throws Exception {
 		activeMQServer.stopJMSBroker();
 	}
 
+	/**
+	 * Check whether the JMS listner has started before complete deployment of the JMS proxy.
+	 * @throws Exception
+	 */
+	@Test(groups = {"wso2.esb"}, description = "Test JMS proxy deployment.")
+	public void testDeploymentOrder() throws Exception {
+
+		JMSQueueMessageProducer sender = new JMSQueueMessageProducer(JMSBrokerConfigurationProvider.getInstance().getBrokerConfiguration());
+		String queueName = "TestQueue";
+		try {
+			sender.connect(queueName);
+			for (int i = 0; i < 5; i++) {
+				sender.pushMessage(message);
+			}
+		} finally {
+			sender.disconnect();
+		}
+
+		Thread.sleep(5000);
+
+		LogViewerClient logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
```

Comment:
Let's retrieve log viewer client  and clear logs before sending the request

Reply:
It is done, check the line 159.

Reply:
Just understood what you trying to test in this test case.